AddQtAndroidApk.cmake

```diff
@@ -97,84 +98,162 @@ macro(add_qt_android_apk TARGET SOURCE_TARGET)
             set(QT_ANDROID_SDK_BUILDTOOLS_REVISION ${BUILD_TOOLS_VERSION})
         endif()
     endforeach()
-    message("Detected Android SDK build tools version ${QT_ANDROID_SDK_BUILDTOOLS_REVISION}")
 
-    # define the application source package directory
-    if(ARG_PACKAGE_SOURCES)
-        set(QT_ANDROID_APP_PACKAGE_SOURCE_ROOT ${ARG_PACKAGE_SOURCES})
+    if(ARG_VERBOSE)
+        message(STATUS "Detected Android SDK build tools version ${QT_ANDROID_SDK_BUILDTOOLS_REVISION}")
+    endif()
+
+    if(ARGS_EXTRA_QML)
+        set(QT_ANDROID_QML_IMPORT_PATH ${ARGS_EXTRA_QML})
+    endif()
```

Comment:
About this new `EXTRA_QML` option: the previous version would scan the whole project tree for QML files; this addition allows to define an explicit path, but is it really necessary? use case?

Reply:
I don't have a use case yet, because i'm using the qtquickcompiler. So all my qmls file are included into the QRC.

Reply:
Great, so let's remove that option.

Reply:
But i'm not sure CMAKE_SOURCE_DIR is also a great default value. Why would it be?
I think main point of that variable is to add Qml modules not in Qt Sdk? 
In your example projects, qml files are included using the .qrc method. In my project i'm using qt quick compiler (so at the end qml files are also in the qrc).
If we remove this variable what would be the way to add an external Qml module?

I'm not a fan of Qml module in general, I prefer to have everything statically linked and embedded into Qrc. But what if someone is using that.

Reply:
> I think main point of that variable is to add Qml modules not in Qt Sdk?

Nope. The `qml-root-path` parameter of `androiddeployqt` is used to scan the project's QML files, in order to detect what plugins and modules they depend on, and copy those libraries to the final APK. So you have to give it a proper value, even if all your QML files are included in the app's qrc. 

`CMAKE_SOURCE_DIR` seems like a good choice because your QML files will most likely be under that root, I can't imagine why one would store its QML files outside the project tree.

Reply:
In my project qml-root-path is set to "" and it still managed to deploy the correct dependencies. (All qml files in the Qt sdk)

Reply:
In mine it didn't work, only some dependencies were found. I had to provide `${CMAKE_SOURCE_DIR}` to make it work. But my project contains two targets: a library and an executable; almost all the QML components are in the library.

Reply:
In my opinion this is the use case of `EXTRA_QML`. To put your folder where you have some extra qml files.
Or you should use the qtquickcompiler from cmake to precompile your file.
In a test project if i put `CMAKE_SOURCE_DIR` as value in the qtdeploy.json file. I end up with bunch of warning because :
* C++ defined modules are not found.
* Qml modules defined in static library are not found
* Private Qml module are not found.
The apk is 600 ko bigger because of useless deployed qml or stuff.
During that test i had in my project directory a `build/` directory that contain the cmake build folder of the app for windows. This folder is under `CMAKE_SOURCE_DIR` so qtandroiddeploy deploy in the apk windows platform related files. Not very clean. 
Ok this is something that happen because i'm in a developement environment . But i don't think the way you set up your Qml dependencies, or the way folder are in your project should impact this tool behavior.
But i also see your point of having a default behavior that work even if not optimized and with warning. But if you want to keep that as a default behavior, we should at least put an option to optimize/remove the warning and useless deployment.